plugins/serializers/wavefront/wavefront.go

```diff
@@ -172,31 +153,62 @@ func buildValue(v interface{}, name string) (float64, error) {
 	}
 }
 
-func formatMetricPoint(metricPoint *wavefront.MetricPoint, s *WavefrontSerializer) []byte {
-	var buffer bytes.Buffer
-	buffer.WriteString("\"")
-	buffer.WriteString(metricPoint.Metric)
-	buffer.WriteString("\" ")
-	buffer.WriteString(strconv.FormatFloat(metricPoint.Value, 'f', 6, 64))
-	buffer.WriteString(" ")
-	buffer.WriteString(strconv.FormatInt(metricPoint.Timestamp, 10))
-	buffer.WriteString(" source=\"")
-	buffer.WriteString(metricPoint.Source)
-	buffer.WriteString("\"")
+func formatMetricPoint(b *buffer, metricPoint *wavefront.MetricPoint, s *WavefrontSerializer) []byte {
+	b.WriteChar('"')
+	b.WriteString(metricPoint.Metric)
+	b.WriteString(`" `)
+	b.WriteFloat64(metricPoint.Value)
+	b.WriteChar(' ')
+	b.WriteUnit64(uint64(metricPoint.Timestamp))
+	b.WriteString(` source="`)
+	b.WriteString(metricPoint.Source)
+	b.WriteChar('"')
 
 	for k, v := range metricPoint.Tags {
-		buffer.WriteString(" \"")
+		b.WriteString(` "`)
 		if s.UseStrict {
-			buffer.WriteString(strictSanitizedChars.Replace(k))
+			b.WriteString(strictSanitizedChars.Replace(k))
 		} else {
-			buffer.WriteString(sanitizedChars.Replace(k))
+			b.WriteString(sanitizedChars.Replace(k))
 		}
-		buffer.WriteString("\"=\"")
-		buffer.WriteString(tagValueReplacer.Replace(v))
-		buffer.WriteString("\"")
+		b.WriteString(`"="`)
+		b.WriteString(tagValueReplacer.Replace(v))
+		b.WriteChar('"')
 	}
 
-	buffer.WriteString("\n")
+	b.WriteChar('\n')
+
+	return *b
+}
+
+// Use a fast and simple buffer for constructing statsd messages
+type buffer []byte
+
+func (b *buffer) Reset() { *b = (*b)[:0] }
+
+func (b *buffer) Copy() []byte {
+	p := make([]byte, len(*b))
+	copy(p, *b)
+	return p
+}
+
+func (b *buffer) WriteString(s string) {
+	*b = append(*b, s...)
+}
+
+// This is named WriteChar instead of WriteByte because the 'stdmethods' check
+// of 'go vet' wants WriteByte to have the signature:
+//
+// 	func (b *buffer) WriteByte(c byte) error { ... }
+//
+func (b *buffer) WriteChar(c byte) {
+	*b = append(*b, c)
+}
+
+func (b *buffer) WriteUnit64(val uint64) {
```

Comment:
function name has a typo in it, which can be misleading (Unit vs Uint)

Reply:
🤦‍♂ thanks - fixed